feeds/pypi/pypi.go

```diff
@@ -73,28 +87,97 @@ func fetchPackages() ([]*Package, error) {
 	return rssResponse.Packages, nil
 }
 
+func fetchCriticalPackages(packageList []string) ([]*Package, error) {
+	responseChannel := make(chan *Response)
```

Comment:
Both of these channels are non buffered, afaict this will lead to the later `select` statements only processing the last value written to them?